app/controllers/editions_controller.rb

```diff
@@ -118,4 +122,25 @@ def gen_guests
     end
   end
 
+  def edition_avg_rating(edition)
+    guest_ids = edition.guests
+    reviews = Review.where(guest_id: guest_ids)
+    sum = 0
```

Comment:
You could avoid initializing the sum variable if you use another iterator (not `each`).

Look at `reduce` if you want. But in this case you could do the sum directly... `sum` is a method that you can call on collections and accept parameters: `reviews.sum(:edition_rating)` probably will be enough ;)

Reply:
I would probably move `edition_avg_rating` and `host_avg_rating` to the `Review` model, and use the same code for both. There's only one line of code that changes! If you create a method that receives the edition and a symbol to know what to sum that's all you need.